src/plan_computation.py

```diff
@@ -59,4 +62,107 @@ def generate_photo_plan_on_grid(
+        for index in range(num_point_x):
+            centers_x.append(offset_x + index * cent_dist_x) #adding edge offset + iterating for each center
+
+    if num_point_y == 1:
+        offset_y = dataset_spec.scan_dimension_y / 2
+        centers_y = [offset_y]
+
+    else:
+        cent_dist_y = min(dist_between_img_y, dataset_spec.scan_dimension_y / gaps_y)
+        offset_y = 0.5 * (dataset_spec.scan_dimension_y - cent_dist_y * gaps_y)
+        for index in range(num_point_y):
+            centers_y.append(offset_y + index * cent_dist_y) 
+
+    flight_plan = []
+    z = dataset_spec.height
+    for index, y in enumerate(centers_y): #enum to help iterate through y coords
+        if index % 2 == 0:
+            x_vals = centers_x
+        else:
+            x_vals = centers_x[::-1] #makes a copy, and then reverses when going up to next row
+        for x in x_vals:
+            flight_plan.append(Waypoint(x, y, z, speed_mps = drone_velocity, photo_trigger = True))
+
+    return flight_plan
+
+def waypoint_times(plan, acc, v_max):
+    """Applies trapezoidal and calculates the time between each waypoint
+
+    Args:
+        plan: reads waypoints, used to iterate through each point for data
+        acc: acceleration of drone
+        v_max: maximum drone speed
+
+    Returns:
+        Time between each waypoint after trapezoidal profiling is applied
+
+    """
+    times = [0.0]
+    time_total = 0.0
+    
+    for inc in range(len(plan) - 1):
+        dist_x = plan[inc + 1].x - plan[inc].x
+        dist_y = plan[inc + 1].y - plan[inc].y
+        dist = math.hypot(dist_x, dist_y)
+    
+        v_blur = plan[inc].speed_mps
+        trap_time, _, _, _ = apply_trapezoidal_prof(dist, acc, v_blur, v_max)
+        time_total += trap_time
+
+        times.append(time_total)
+    return times
+
+def apply_trapezoidal_prof(dist, acc, v_blur, v_max):
+    """Calculates the trapezoidal profiling for the drone based on 4 inputs
+
+    Args:
+        dist: distance from one image point to another
+        acc: acceleration of drone
+        v_blur: calculated drone velocity for acceptable blur
+        v_max: maximum drone speed
+
+    Returns:
+        Time between each waypoint after trapezoidal profiling is applied
+
+    """
+    s_acc = (v_max**2 - v_blur**2) / (2 * acc)
+    if 2 * s_acc <= dist: #if a trapezoid
+        t_acc = (v_max - v_blur) / acc
+        t_cruise = (dist - 2 * s_acc) / v_max
+        t_total = 2 * t_acc + t_cruise
+
+        times = [0.0, t_acc, t_acc + t_cruise, t_total]
+        vels  = [v_blur, v_max, v_max, v_blur]
+        shape = "Trapezoid"
+
+    else: #if a triangle
+        v_peak = math.sqrt(v_blur**2 + acc * dist)
+        t_acc = (v_peak - v_blur) / acc
+        t_total = 2 * (v_peak - v_blur) / acc
+        
+        times = [0.0, t_acc, t_total]
+        vels  = [v_blur, v_peak, v_blur]
+        shape = "Triangle"
+    
+    return t_total, times, vels, shape
```

Comment:
Just noting there that the number of entries in times and vels can change depending on the profile. I hope you handle it correctly at the caller.